packages/kit/src/runtime/client/router.js

```diff
@@ -157,16 +157,16 @@ export class Router {
 
 			if (!this.owns(url)) return;
 
-			const noscroll = a.hasAttribute('sveltekit:noscroll');
+			// Check if new url only differs by hash
+			if (url.href.split('#')[0] === location.href.split('#')[0]) {
+				// Delay pushState, otherwise the browser default behaviour would not kick in
+				setTimeout(() => history.pushState({}, '', url.href));
```

Comment:
If we're returning without preventing the event default, do we still need to push state?

Reply:
Yeah. The `pushState` is so that we keep a new history entry of the new url with hash so that we can go back to it. There were tests for this behaviour specifically. I forgot to add a comment regarding this though.

Reply:
Updated in 79350cb